batch7crud-roll8/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDao.java

```diff
@@ -0,0 +1,182 @@
+    Employee employee = null;
+    ResultSet resultSet = null;
+
+    try (Connection conn = DBConnection.getConnection();
+      PreparedStatement statement = conn.prepareStatement(EMPLOYEE_SELECT_BY_ID)
+    ){
+      statement.setInt(1, id);
+      resultSet = statement.executeQuery();
+      if (resultSet.next()) {
+        employee = new Employee();
+        employee.setId(resultSet.getInt(ModelConstants.ID));
+        employee.setName(resultSet.getString(ModelConstants.NAME));
+        employee.setAddress(resultSet.getString(ModelConstants.ADDRESS));
+        employee.setDesignation(resultSet.getString(ModelConstants.DESIGNATION));
+        employee.setPhone(resultSet.getString(ModelConstants.PHONE));
+      }
+      return employee;
+    }
+    catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+    finally {
+      try {
+        if (resultSet != null)
+          resultSet.close();
+      } catch (Exception e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        throw new DataException(e.getMessage());
+      }
+    }
+
+  }
+
+  public void edit(Employee employee) throws DataException {
+    try (Connection conn = DBConnection.getConnection();
+      PreparedStatement statement = conn.prepareStatement(EMPLOYEE_UPDATE_BY_ID)
+    ){
+      statement.setString(1, employee.getName());
+      statement.setString(2, employee.getAddress());
+      statement.setString(3, employee.getDesignation());
+      statement.setString(4, employee.getPhone());
+      statement.setInt(5, employee.getId());
+      statement.executeUpdate();
+    }
+    catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void delete(Integer id) throws DataException {
+    try (Connection conn = DBConnection.getConnection();
+      PreparedStatement statement = conn.prepareStatement(EMPLOYEE_DELETE_BY_ID)
+    ) {
+      statement.setInt(1, id);
+      statement.executeUpdate();
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public Integer count() throws DataException {
+    ResultSet resultSet = null;
+    try(Connection conn = DBConnection.getConnection();
+        PreparedStatement statement = conn.prepareStatement(EMPLOYEE_GET_COUNT)
+    ){
+      resultSet = statement.executeQuery();
+      if (resultSet.next()){
+        return resultSet.getInt(ModelConstants.TOTAL);
+      }
+      else {
+        return 0;
+      }
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+    finally {
+      try {
+        if (resultSet != null)
+          resultSet.close();
+      } catch (Exception e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        throw new DataException(e.getMessage());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not throw exceptions in finally blocks. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1163)